src/message.c

```diff
@@ -1039,6 +1035,8 @@ add_msg_hist(
     if (first_msg_hist == NULL)
 	first_msg_hist = last_msg_hist;
     ++msg_hist_len;
+
+    check_msg_hist();
```

Comment:
why was this moved down here?

Reply:
If we leave it in its original position, `&msghistory + 1` records will be remembered, right?

Reply:
In the previous implementation, the constant 200 actually remembered 201 records.

Reply:
ah yes correct

Reply:
Yes.  The previous implementation is not correct.